ethash/src/compute.rs

```diff
@@ -43,6 +44,7 @@ pub struct ProofOfWork {
 pub struct Light {
 	block_number: u64,
 	cache: NodeCache,
+	c_dag: Option<CDag>,
```

Comment:
This could be an enum like 'EthashVariant', if there is some progpow2 or other. Yet it means that it needs to be build initiated from a fn parameter in new_with_builder.
I don't know if it is worth the added complexity (enum makes it explicit that c_dag condition usage of progpow but it may be better to keep it simple).